### PR TITLE
Media: slightly better errors for failed uploads

### LINKED ISSUE #6562: Media - Better error messaging on failed uploads
We should provide user friendly, non-technical, error messages in case of uploading errors. 

I know this is a really wide ticket, that has its roots in a lot of parts of the app, but we should try to avoid error messages that implies technical skills. 
Below is an example:

Most users have no idea what the message `The request timed out means`.

Another example of technical language:

**Required**  I think we need to translate the most common error messages.

**Bonus**: Provide a way (maybe link with instructions?) on how to resolve the underlaying problem on the server.

### PR DESCRIPTION
**Pull request by kurzee**
Fixes #6562

A small PR to get us in better place for error messaging when a media upload fails. For the most part, we're just overriding the raw `NSURLError` domain messages with something that encourages the user to try again. For self-hosted though, we go ahead and pass on the raw value except for two previous cases. This way self-hosted can debug and REST can try again.

When we refactor for async uploading we'll have more opportunity to really inform users of what went wrong and where.

To test:
1. Generate the error messages by selecting a media to insert and have it fail.
2. You can easily test the `Media upload failed, internet connection appears to be offline.` by going into airplane mode.
3. The other cases are a bit more difficult and I couldn't find away to trigger them, instead I suggest changed the error code to test the messages.

Needs review: @frosty

### CODE REVIEW

File: WordPress/Classes/Services/MediaService.m [hunk 1]
```diff
@@ -356,24 +356,49 @@ - (void)updateMultipleMedia:(NSArray<Media *> *)mediaObjects
...
+            switch (error.code) {
+                case NSURLErrorUnknown:
+                    // Unknown error, encourage the user to try again
+                    // Note: if support requests show up with this error message we can rule out known NSURLError codes
+                    customErrorMessage = NSLocalizedString(@"Media upload failed, an unknown error occurred, please try again.", @"Error message shown when a media upload fails for unknown reason and the user should try again.");
+                    break;
+                case NSURLErrorNetworkConnectionLost:
+                case NSURLErrorNotConnectedToInternet:
+                    // Clear lack of device internet connection, notify the user
+                    customErrorMessage = NSLocalizedString(@"Media upload failed, internet connection appears to be offline.", @"Error message shown when a media upload fails because the user isn't connected to the internet.");
```
Comment (frosty) [inline WordPress/Classes/Services/MediaService.m]: See my other comment about "Media upload failed" – perhaps this should just be "The internet connection appears to be offline."

File: WordPress/Classes/Services/MediaService.m [hunk 2]
```diff
@@ -356,24 +356,49 @@ - (void)updateMultipleMedia:(NSArray<Media *> *)mediaObjects
...
-        newError = [[NSError alloc] initWithDomain:WPXMLRPCFaultErrorDomain code:errorCode userInfo:userInfo];
+    } else if ([remote isKindOfClass:[MediaServiceRemoteREST class]]) {
+        // For WPCom/Jetpack sites and NSURLErrors we should use a more general error message to encourage a user to try again,
+        // rather than raw NSURLError messaging.
+        if ([error.domain isEqualToString:NSURLErrorDomain]) {
+            switch (error.code) {
+                case NSURLErrorUnknown:
+                    // Unknown error, encourage the user to try again
+                    // Note: if support requests show up with this error message we can rule out known NSURLError codes
+                    customErrorMessage = NSLocalizedString(@"Media upload failed, an unknown error occurred, please try again.", @"Error message shown when a media upload fails for unknown reason and the user should try again.");
```
Comment (frosty) [inline WordPress/Classes/Services/MediaService.m]: See my other comment about "Media upload failed" – perhaps this should just be "An unknown error occurred. Please try again."

File: WordPress/Classes/Services/MediaService.m [hunk 3]
```diff
@@ -356,24 +356,49 @@ - (void)updateMultipleMedia:(NSArray<Media *> *)mediaObjects
...
+                    break;
+                case NSURLErrorNetworkConnectionLost:
+                case NSURLErrorNotConnectedToInternet:
+                    // Clear lack of device internet connection, notify the user
+                    customErrorMessage = NSLocalizedString(@"Media upload failed, internet connection appears to be offline.", @"Error message shown when a media upload fails because the user isn't connected to the internet.");
+                    break;
+                default:
+                    // Default NSURL error messaging, probably server-side, encourage user to try again
+                    customErrorMessage = NSLocalizedString(@"Media upload failed, please try again in a moment.", @"Error message shown when a media upload fails for a general network issue and the user should try again in a moment.");
```
Comment (frosty) [inline WordPress/Classes/Services/MediaService.m]: See my other comment about "Media upload failed" – perhaps this should just be "Something went wrong. Please try again."

**Comment by kurzee** [comment]
@frosty changed my mind 😄 , I like the short form better after seeing it.

**Comment by frosty** [comment]
:D :shipit:!

**Pull request merged**